path_oidc.go

```diff
@@ -0,0 +1,303 @@
+	allClaims, err := b.verifyToken(ctx, config, role, rawToken)
+	if err != nil {
+		return logical.ErrorResponse("%s %s", errTokenVerification, err.Error()), nil
+	}
+
+	// Attempt to fetch information from the /userinfo endpoint and merge it with
+	// the existing claims data. A failure to fetch additional information from this
+	// endpoint will not invalidate the authorization flow.
+	if userinfo, err := provider.UserInfo(ctx, oauth2.StaticTokenSource(oauth2Token)); err == nil {
+		_ = userinfo.Claims(&allClaims)
+	} else {
+		logFunc := b.Logger().Warn
+		if strings.Contains(err.Error(), "user info endpoint is not supported") {
+			logFunc = b.Logger().Info
+		}
+		logFunc("error reading /userinfo endpoint", "error", err)
+	}
+
+	if allClaims["nonce"] != state.nonce {
+		return logical.ErrorResponse(errTokenVerification + " Invalid ID token nonce."), nil
+	}
+	delete(allClaims, "nonce")
+
+	alias, groupAliases, err := b.createIdentity(allClaims, role)
+	if err != nil {
+		return logical.ErrorResponse(err.Error()), nil
+	}
+
+	tokenMetadata := map[string]string{"role": roleName}
+	for k, v := range alias.Metadata {
+		tokenMetadata[k] = v
+	}
+
+	resp := &logical.Response{
+		Auth: &logical.Auth{
+			Policies:     role.Policies,
+			DisplayName:  alias.Name,
+			Period:       role.Period,
+			NumUses:      role.NumUses,
+			Alias:        alias,
+			GroupAliases: groupAliases,
+			InternalData: map[string]interface{}{
+				"role": roleName,
+			},
+			Metadata: tokenMetadata,
+			LeaseOptions: logical.LeaseOptions{
+				Renewable: true,
+				TTL:       role.TTL,
+				MaxTTL:    role.MaxTTL,
+			},
+			BoundCIDRs: role.BoundCIDRs,
+		},
+	}
+
+	return resp, nil
+}
+
+// authURL returns a URL used for redirection to receive an authorization code.
+// This path requires a role name, or that a default_role has been configured.
+// Because this endpoint is unauthenticated, the response to invalid or non-OIDC
+// roles is intentionally non-descriptive and will simply be an empty string.
+func (b *jwtAuthBackend) authURL(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	logger := b.Logger()
+
+	// default response for most error/invalid conditions
+	resp := &logical.Response{
+		Data: map[string]interface{}{
+			"auth_url": "",
+		},
+	}
+
+	config, err := b.config(ctx, req.Storage)
+	if err != nil {
+		logger.Warn("error loading configuration", "error", err)
+		return resp, nil
+	}
+
+	if config == nil {
+		logger.Warn("nil configuration")
+		return resp, nil
+	}
+
+	roleName := d.Get("role").(string)
+	if roleName == "" {
+		roleName = config.DefaultRole
+	}
+	if roleName == "" {
+		return logical.ErrorResponse("missing role"), nil
+	}
+
+	redirectURI := d.Get("redirect_uri").(string)
+	if redirectURI == "" {
+		return logical.ErrorResponse("missing redirect_uri"), nil
+	}
+
+	role, err := b.role(ctx, req.Storage, roleName)
+	if err != nil {
+		logger.Warn("error loading role", "error", err)
```

Comment:
I don't think we need to log here or below

Reply:
Removed